src/js/frontend/src/components/login/LoginForm.tsx

```diff
@@ -0,0 +1,63 @@
+import React from "react";
+import { Field, Form, Formik } from "formik";
+import { GoogleSSO } from "./GoogleSSO";
+
+import { loginService } from "../../services/login/loginService";
+import { yupValidationSchema } from "./yupValidationSchema";
+
+const LoginForm = () => {
+  const [state, setState] = React.useState({
+    loginFailed: false, // Boolean represented as true when user is successfully authenticated,
+    // false when there's an auth error, token has been removed or user has been logged out
+  });
+
+  const handleSubmit = async (values: { username: string; password: string }) => {
+    const loginSuccess = await loginService(values.username, values.password);
```

Comment:
Can we meet in the middle with `successfulLoginResponse`? Because it's a boolean, I think it makes sense to name the variable based on the state of what's true and what's false.`loginResult` set to a bool is slightly confusing because it can return an object with login values or an error, but I'm happy to change it to `loginResult` if unclear.

Reply:
Reading more of the code, I think that my issue ultimately lies in the fact that our `loginService` function is returning a boolean, rather than the response itself. Because of that, we now have a lot of booleans in this scope that sound like they're the same thing, but actually aren't, which is really confusing.

If that function just returned the request, then _it_ would be simpler (it's just making a request, not making a request and figuring out what to do with it), and the code _here_ would be more clear:

```typescript
const loginResponse = await loginService(values.username, values.password);

if loginResponse["success"] {
    // do successful things
} else {
    // do failure things, including operating on any error messages that may be in the response!
    loginStatus = false;
}
```
To my eye, this reads much more clearly, and there is no ambiguity. Additionally, this formulation is _more useful_, because any error messages in a failed login response are not swallowed by the `loginService` code.

(Incidentally, `loginService.tsx` is identical to `signInWithGoogleService.ts`... I assume the latter should be deleted?)